### PR TITLE
GLES: Implement shader based 111R swizzle

### PR DESCRIPTION
**Pull request by sarbes**
## Description
This PR implements shader based 111R swizzling for GLES 2.0 systems.

## Motivation and context
Previously on GLES 2.0 systems, alpha only textures (swizzle 111R) would have to be converted to GL_LUMINANCE_ALPHA in order to be displayed. This took a bit of time when uploading the texture. It would also take up twice the memory on the GPU, as well as twice the memory bandwidth to render each frame.

Alpha only textures are widely used in skins, so this is a major gain. It will also allow ETC1 textures with alpha packed in the red channel.

## How has this been tested?
When forcing GLES 2.0, Estuary still renders fine on my system. Renderdoc confirms the GL_LUMINANCE textures.

## What is the effect on users?
Less jank and less resource usage on GLES 2.0.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by sarbes** [comment]
Whoops.

I've renamed the shaders and enums to make the swizzle more clear.

File: xbmc/guilib/GUITextureGLES.cpp [hunk 1]
```diff
@@ -65,33 +67,62 @@ void CGUITextureGLES::Begin(KODI::UTILS::COLOR::Color color)
...
+    else
+    {
+      m_renderSystem->EnableGUIShader(ShaderMethodGLES::SM_MULTI);
+    }
 
     hasAlpha |= m_diffuse.m_textures[0]->HasAlpha();
 
-    m_diffuse.m_textures[0]->BindToUnit(1);
-
+    // We don't have ALPHA_COLOR versions of the GLES 2.0 shaders, so in the
```
Comment (sarbes) [inline xbmc/guilib/GUITextureGLES.cpp:96]: On further thought, I need to change the UV coordinates too when doing this.

**Review by neo1973 — APPROVED** [review]
Can confirm that all textures are displaying correctly on a RPi3B.

**Comment by sarbes** [comment]
jenkins build this please

**Pull request merged**